### PR TITLE
Warrants pt 2 - Allow multiple sys validation dependencies

### PR DESCRIPTION
**Pull request by maackle**
### Summary

Part of #3639 

With the addition of ChainFork warrants, with two actions referenced, DhtOps in general now can have multiple sys validation dependencies.

However now that i wrote this, this function is only being used to check for the existence of dependencies, the actual deps returned are never used.

### TODO:
- [ ] CHANGELOGs updated with appropriate info

### CODE REVIEW

File: crates/hc_sleuth/src/event.rs [hunk 1]
```diff
@@ -199,15 +199,17 @@ impl aitia::Fact for Event {
...
                 let fetched = Fetched {
                     by: by.clone(),
                     op: op.clone(),
                 }
                 .into();
 
-                if let Some(dep) = dep {
+                // TODO: this must be generalized to support multiple dependencies.
+                //       we're only using the first here.
+                if let Some(dep) = dep.first() {
```
Comment (maackle) [inline crates/hc_sleuth/src/event.rs:212]: This only affects hc_sleuth which is an experimental diagnostic tool

File: crates/holochain_sqlite/src/sql/cell/update_dep_create_link.sql [hunk 2]
```diff
@@ -11,11 +11,14 @@ WHERE
...
       1
     FROM
-      DhtOp AS OP_DEP
+      DhtOp AS DhtOpDep
     WHERE
-      OP_DEP.action_hash = DhtOp.dependency
-      AND OP_DEP.when_integrated IS NOT NULL
-      AND OP_DEP.type = :create_link
+      (
+        DhtOpDep.action_hash = DhtOp.dependency
```
Comment (jost-s) [inline crates/holochain_sqlite/src/sql/cell/update_dep_create_link.sql:17]: Is it possible to have the convention to store chain op deps in `DhtOp.dependency` and warrant op dependency 2 in `DhtOp.dependency2`? This change wouldn't be needed then.

Comment (maackle) [reply crates/holochain_sqlite/src/sql/cell/update_dep_create_link.sql:17]: I'm not sure what you mean. The way I read this, it sounds like you're suggesting putting a chain op dep in one slot, and one of the warrant op deps in the other?

Comment (jost-s) [reply crates/holochain_sqlite/src/sql/cell/update_dep_create_link.sql:17]: Sorry, yes, exactly. Keep the first slot `dependency` for chain ops' sys dependency or a warrant op's first dependency accordingly, and the second slot only for the warrant op's second dependency.

Comment (maackle) [reply crates/holochain_sqlite/src/sql/cell/update_dep_create_link.sql:17]: Oh yes of course. As you say it is how it is. And I see that the comment you're making is that all of these queries are only for chain ops, which have no second dependency, and so including that is redundant, which is also correct. I'll change that back to only check the first dep.

**Review by jost-s — APPROVED** [review]
LGTM

**Review by ThetaSinner — APPROVED** [review]

**Pull request merged**